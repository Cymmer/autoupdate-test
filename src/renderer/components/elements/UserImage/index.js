import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

import Placeholder from '../../../static/images/placeholder.svg';

const UserImage = ({ className, imageClassName, image, id }) => (
  <div className={className}>
    <img
      alt="User"
      className={cn(styles.UserImage_image, imageClassName)}
      data-test="image"
      id={id}
      src={image || Placeholder}
    />
  </div>
);

UserImage.defaultProps = {
  image: null,
  className: null,
  imageClassName: null,
  id: null,
};

UserImage.propTypes = {
  image: PropTypes.any,
  className: PropTypes.string,
  imageClassName: PropTypes.string,
  id: PropTypes.string,
};

export default UserImage;
