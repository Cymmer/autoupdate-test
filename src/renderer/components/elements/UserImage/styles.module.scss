.UserImage {
  &_image {
    width: 100%;
    border-radius: 100%;
  }
}
