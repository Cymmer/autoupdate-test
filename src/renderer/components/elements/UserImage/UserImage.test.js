import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from 'codechum-app-utils';
import UserImage from './index';

import Placeholder from '../../../static/images/placeholder.svg';

describe('RENDERING - UserImage', () => {
  it('should render with default props', () => {
    const component = shallow(<UserImage />);
    expect(component).toMatchSnapshot();
  });

  it('should render the image correctly', () => {
    let component = shallow(<UserImage />);
    expect(findByTestAttr(component, 'image').prop('src')).toBe(Placeholder);

    const overrideImage = 'test.png';
    component = shallow(<UserImage image={overrideImage} />);
    expect(findByTestAttr(component, 'image').prop('src')).toBe(overrideImage);
  });
});
