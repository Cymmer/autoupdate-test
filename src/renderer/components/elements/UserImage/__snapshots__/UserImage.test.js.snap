// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - UserImage should render with default props 1`] = `
<div
  className={null}
>
  <img
    alt="User"
    className="UserImage_image"
    data-test="image"
    id={null}
    src="test-file-stub"
  />
</div>
`;
